### PR TITLE
repo: default to target arch for stage package cache

### PR DESCRIPTION
**Pull request by cjp256**

### CODE REVIEW

File: snapcraft/internal/repo/apt_cache.py [hunk 1]
```diff
@@ -38,13 +38,20 @@
...
+        self,
+        *,
+        stage_cache: Optional[Path] = None,
+        stage_cache_arch: Optional[str] = None,
+    ) -> None:
         self.stage_cache = stage_cache
+        self.stage_cache_arch = stage_cache_arch
 
     def __enter__(self) -> "AptCache":
+
```
Comment (sergiusens) [inline snapcraft/internal/repo/apt_cache.py:51]: why?

**Review by sergiusens — COMMENTED** [review]
Interesting, AFAIK, ros does not support cross compilation

File: snapcraft/internal/repo/apt_cache.py (hunk 1 above)
Comment (cjp256) [inline snapcraft/internal/repo/apt_cache.py:51]:
```suggestion
```

Comment (cjp256) [reply snapcraft/internal/repo/apt_cache.py:51]: typo, fixed

**Comment by cjp256** [comment]
> Interesting, AFAIK, ros does not support cross compilation

I don't think it does either - but for consistency's sake, I've included it.  I would expect target arch is always the host arch in that case.

**Comment by cjp256** [comment]
FWIW, I'm going to add some spread test to cover this as well.

**Review by sergiusens — APPROVED** [review]

**Pull request merged**